homeassistant/components/volumio/media_player.py

```diff
@@ -68,91 +53,59 @@
 
 PLAYLIST_UPDATE_INTERVAL = timedelta(seconds=15)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    }
-)
-
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Comment:
Should we keep `async_setup_platform` and do something like `harmony/remote.py` to avoid the breaking change?

```
    if discovery_info:
        # Now handled by ssdp in the config flow
        return

    if find_matching_config_entries_for_host(hass, config[CONF_HOST]):
        return

    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": SOURCE_IMPORT}, data=validated_config
        )
    )
```

Reply:
I thought about it, but realized that the existing integration automatically creates an entity, and unlike other integrations, that aren't any parameters that are only available through manual configuration. This led me to believe that:
1. It'll already be a breaking change for most users (automatic discovery vs needing to go through a discovery config flow)
2. For the minority with manual configuration, I'm not sure it's worth the extra complexity, especially given that w/o discovery getting a unique id is broken (this will be fixed in the next Volumio version, and my code handles the error).